ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java

```diff
@@ -185,14 +185,10 @@ private void handleAndLogImportFailure(
           invalidBlock.toLogString(),
           result.causedBy().get());
       LOG.debug("with stack", result.causedBy().get());
-      if (!result.isInternalError()) {
-        badBlockManager.addBadBlock(invalidBlock);
-      }
-
     } else {
       LOG.info("{}. Block {}", result.errorMessage, invalidBlock.toLogString());
-      badBlockManager.addBadBlock(invalidBlock);
     }
+    badBlockManager.addBadBlock(invalidBlock, result.causedBy());
```

Comment:
before an internal error did not cause the block to be added to the bad list, now it is always added, is this wanted?